kiwi/bootloader/config/base.py

```diff
@@ -298,12 +298,15 @@ def get_boot_path(self, target='disk'):
                 filesystem = self.xml_state.build_type.get_filesystem()
                 volumes = self.xml_state.get_volumes()
                 if filesystem == 'btrfs' and volumes:
-                    # boot or boot/grub2 on a subvolume prevents grub from
-                    # finding its config file
+                    # grub boot data paths must not be in a subvolume
+                    # otherwise grub won't be able to find its config file
+                    grub_boot_data_paths = ['boot', 'boot/grub', 'boot/grub2']
                     for volume in volumes:
-                        if volume.name == 'boot' or volume.name == 'boot/grub2':
+                        if volume.name in grub_boot_data_paths:
```

Comment:
I would also suggest naming the variable `grub2_boot_data_paths` to make it clear grub2 is the only thing supported.